main/src/ca/uwaterloo/flix/language/phase/Typer.scala

```diff
@@ -256,13 +256,36 @@ object Typer extends Phase[KindedAst.Root, TypedAst.Root] {
                 case Validation.Failure(errs) =>
                   val instanceErrs = errs.collect {
                     case UnificationError.NoMatchingInstance(tconstr) =>
-                      TypeError.NoMatchingInstance(tconstr.sym, tconstr.arg, tconstr.loc)
+                      if (tconstr.sym.name == "Eq")
+                        TypeError.MissingEq(tconstr.arg, tconstr.loc)
```

Comment:
Is there a risk that someone has their own local `Eq` class? I assume there's a name clash but it might be better to check more than just the name.

Reply:
Yes, but I don't know how to get a hold of the ClassSym at this point in the code.